EIPS/eip-2579.md

```diff
@@ -0,0 +1,128 @@
+---
+eip: 2579
+title: Merkle Tree Generalized Index
+author: naxe (ngmachado @LogicB0x)
+discussions-to: https://ethereum-magicians.org/t/merkle-tree-generalized-index/4162/
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-03-31
+---
+
+## Simple Summary
+
+Using a Merkle Tree with a Generalized Index coordination system as used in other systems with the necessary adaptation to a smart contract environment, reducing the computation requirement and storage space.
+
+## Motivation
+
+Bring the capacity to the developer community to implement a selective proof or multi proof system “as a service” in the form of a library, helping the intercommunication between different ecosystems in Ethereum space by using the same library to process profs.
```

Comment:
The motivation is kind of vague at the moment.  Perhaps you can give a bit more of a concrete use case as an example of what problem this solves?

Reply:
The motivation also should include reasoning why this needs to be standardized.  Keep in mind that not all Good Ideas need to be Standards.  Object Oriented Programming or Functional Programming may be Good Ideas, but they are not standardized.  Same with algorithms, there are many great algorithms out there and using them is a Good Idea, but they don't need a Standard.

Usually a standard is necessary when you have a many-to-many relationship.  Browsers are a great example of this where you have many people creating websites and many people creating browsers and we want all the websites to work with all of the browsers.  If there was a single browser, then a standard would not be necessary because that is a one to many relationship.